keras/callbacks.py

```diff
@@ -1232,3 +1234,72 @@ def __init__(self,
+
+       If save_best_only is True, saves the MXNet model in the format '<prefix>-symbol.json' and '<prefix>-0000.params'.
+       i.e., uses 0000 as placeholder for the epoch. You will have only one best model at the end of training.
+
+       If save_best_only is False, i.e., you want to save the MXNet model after each epoch, this callback saves the Model
+       in the format '<prefix>-symbol.json' and '<prefix>-<epoch>.params'. Example: If you are running the training job
+       for 2 epochs, you will have one symbol file - '<prefix>-symbol.json' and 2 params file, one each for the 2 epochs,
+       '<prefix>-0001.params' and '<prefix>-0002.params'
+
+       # Arguments
+           prefix: Prefix name of the saved MXNet Model (symbol and params) files.
+                   Model will be saved as '<prefix>-symbol.json' and '<prefix>-<epoch>.params'.
+           monitor: quantity to monitor.
+           verbose: verbosity mode, 0 or 1.
+           save_best_only: if `save_best_only=True`,
+               the latest best model according to
+               the quantity monitored will not be overwritten.
+           mode: one of {auto, min, max}.
+               If `save_best_only=True`, the decision
+               to overwrite the current save file is made
+               based on either the maximization or the
+               minimization of the monitored quantity. For `val_acc`,
+               this should be `max`, for `val_loss` this should
+               be `min`, etc. In `auto` mode, the direction is
+               automatically inferred from the name of the monitored quantity.
+           period: Interval (number of epochs) between checkpoints.
+       """
+    def __init__(self, prefix, monitor='val_loss', verbose=0,
+                 save_best_only=False,
+                 mode='auto', period=1):
+        super(MXNetModelCheckpoint, self).__init__(filepath=None, monitor=monitor, verbose=verbose,
+                                                   save_best_only=save_best_only, save_weights_only=False,
+                                                   mode=mode, period=period)
+        self.prefix = prefix
+
+    def on_epoch_end(self, epoch, logs=None):
+        logs = logs or {}
+        self.epochs_since_last_save += 1
+        if self.epochs_since_last_save >= self.period:
+            self.epochs_since_last_save = 0
+            if self.save_best_only:
+                # We use epoch=0 for saving the best model. This will replace previous best model.
+                current = logs.get(self.monitor)
+                if current is None:
+                    warnings.warn('Can save best model only with %s available, '
+                                  'skipping.' % self.monitor, RuntimeWarning)
+                else:
+                    if self.monitor_op(current, self.best):
+                        if self.verbose > 0:
```

Comment:
self.verbose ==1 - is verbose expected to have multiple values? or only between 0 and 1?